semgrep/tests/e2e/test_ignores.py

```diff
@@ -51,3 +51,19 @@ def test_file_not_relative_to_base_path(tmp_path, monkeypatch, snapshot):
     )
     stdout, _ = process.communicate("a")
     snapshot.assert_match(_clean_output_json(stdout), "results.json")
+
+
+def test_internal_explicit_semgrepignore(run_semgrep_in_tmp, tmp_path, snapshot):
+
+    (tmp_path / ".semgrepignore").symlink_to(
```

Comment:
looking at the test, it's hard to see that it works because I need to read an extra file for the code, an extra file for the rule, and then the snapshot to see if it match. There has to be a more declarative way to do.

Reply:
I agree that the e2e tests are hard to read. The tests in `unit/` are closer to what you want, since they're just testing individual bits of code. It's not entirely clear to me how we'd make e2e tests better, because by definition they have to involve actually calling Semgrep and then looking at the whole output, which contains a bunch of text. The test I added should have had a comment on it though.